includes/fields/class-attachment-upload.php

```diff
@@ -243,6 +243,13 @@ protected function sanitize() {
 
 		if ( empty( $this->value ) ) {
 			$this->value = null;
+		} elseif ( absint( get_option( 'hp_installed_time' ) ) > strtotime( '12 June 2024' ) ) {
+			$this->value = Models\Attachment::query()->filter(
+				[
+					'id__in'       => (array) $this->value,
+					'parent_field' => hp\unprefix( $this->get_name() ),
```

Comment:
We seem to have this issue for images only, because if you create different Attachment attributes they are not mixed up. Please test this with the listing images, upload an image via the Add Media button above the back-end listing description editor (not via our Images uploader) and check if images appear in the listing gallery or in the front-end Images uploader (with this coded solution temporarily disabled).

Reply:
Yes, if upload images into listing on the backend with Add Media button the bug appears. If add images from Media Library (images which are already uploaded) then bug does not appears

Reply:
This may also require a fix here https://github.com/hivepress/hivepress/blob/master/includes/models/class-listing.php#L219 because there's a core get_attached_media function and it fetches all the images (including those uploaded via the Add Media button above the editor), it should also filter images by parent field name.